typing/jkind.mli

```diff
@@ -821,6 +834,14 @@ val sub_or_error :
   ('l * allowed) Types.jkind ->
   (unit, Violation.t) result
 
+(** Compares the layouts of two jkinds.
+*)
+val sub_jkind_l_layout :
```

Comment:
If it's just looking at layouts, I think it shouldn't need the `jkind_context` or to care about `l` vs `r`.